Index: src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java
===================================================================
--- src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java	(revision 44067f66e66df23f8edb88112d3e152c779e0c38)
+++ src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java	(revision 44067f66e66df23f8edb88112d3e152c779e0c38)
@@ -1,31 +0,0 @@
-package ru.javawebinar.topjava.util;
-
-import ru.javawebinar.topjava.model.UserMeal;
-import ru.javawebinar.topjava.model.UserMealWithExceed;
-
-import java.time.LocalDateTime;
-import java.time.LocalTime;
-import java.time.Month;
-import java.util.Arrays;
-import java.util.List;
-
-public class UserMealsUtil {
-    public static void main(String[] args) {
-        List<UserMeal> mealList = Arrays.asList(
-                new UserMeal(LocalDateTime.of(2015, Month.MAY, 30,10,0), "Завтрак", 500),
-                new UserMeal(LocalDateTime.of(2015, Month.MAY, 30,13,0), "Обед", 1000),
-                new UserMeal(LocalDateTime.of(2015, Month.MAY, 30,20,0), "Ужин", 500),
-                new UserMeal(LocalDateTime.of(2015, Month.MAY, 31,10,0), "Завтрак", 1000),
-                new UserMeal(LocalDateTime.of(2015, Month.MAY, 31,13,0), "Обед", 500),
-                new UserMeal(LocalDateTime.of(2015, Month.MAY, 31,20,0), "Ужин", 510)
-        );
-        getFilteredWithExceeded(mealList, LocalTime.of(7, 0), LocalTime.of(12,0), 2000);
-//        .toLocalDate();
-//        .toLocalTime();
-    }
-
-    public static List<UserMealWithExceed>  getFilteredWithExceeded(List<UserMeal> mealList, LocalTime startTime, LocalTime endTime, int caloriesPerDay) {
-        // TODO return filtered list with correctly exceeded field
-        return null;
-    }
-}
Index: src/main/java/ru/javawebinar/topjava/util/MealsUtil.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/util/MealsUtil.java	(date 1530137160353)
+++ src/main/java/ru/javawebinar/topjava/util/MealsUtil.java	(date 1530137160353)
@@ -0,0 +1,31 @@
+package ru.javawebinar.topjava.util;
+
+import ru.javawebinar.topjava.model.Meal;
+import ru.javawebinar.topjava.model.MealWithExceed;
+
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.Month;
+import java.util.Arrays;
+import java.util.List;
+
+public class MealsUtil {
+    public static void main(String[] args) {
+        List<Meal> mealList = Arrays.asList(
+                new Meal(LocalDateTime.of(2015, Month.MAY, 30,10,0), "Завтрак", 500),
+                new Meal(LocalDateTime.of(2015, Month.MAY, 30,13,0), "Обед", 1000),
+                new Meal(LocalDateTime.of(2015, Month.MAY, 30,20,0), "Ужин", 500),
+                new Meal(LocalDateTime.of(2015, Month.MAY, 31,10,0), "Завтрак", 1000),
+                new Meal(LocalDateTime.of(2015, Month.MAY, 31,13,0), "Обед", 500),
+                new Meal(LocalDateTime.of(2015, Month.MAY, 31,20,0), "Ужин", 510)
+        );
+        getFilteredWithExceeded(mealList, LocalTime.of(7, 0), LocalTime.of(12,0), 2000);
+//        .toLocalDate();
+//        .toLocalTime();
+    }
+
+    public static List<MealWithExceed>  getFilteredWithExceeded(List<Meal> mealList, LocalTime startTime, LocalTime endTime, int caloriesPerDay) {
+        // TODO return filtered list with correctly exceeded field
+        return null;
+    }
+}
\ No newline at end of file
Index: src/main/java/ru/javawebinar/topjava/model/UserMealWithExceed.java
===================================================================
--- src/main/java/ru/javawebinar/topjava/model/UserMealWithExceed.java	(revision 44067f66e66df23f8edb88112d3e152c779e0c38)
+++ src/main/java/ru/javawebinar/topjava/model/MealWithExceed.java	(date 1530137160292)
@@ -2,7 +2,7 @@
 
 import java.time.LocalDateTime;
 
-public class UserMealWithExceed {
+public class MealWithExceed {
     private final LocalDateTime dateTime;
 
     private final String description;
@@ -11,7 +11,7 @@
 
     private final boolean exceed;
 
-    public UserMealWithExceed(LocalDateTime dateTime, String description, int calories, boolean exceed) {
+    public MealWithExceed(LocalDateTime dateTime, String description, int calories, boolean exceed) {
         this.dateTime = dateTime;
         this.description = description;
         this.calories = calories;
Index: src/main/java/ru/javawebinar/topjava/model/UserMeal.java
===================================================================
--- src/main/java/ru/javawebinar/topjava/model/UserMeal.java	(revision 44067f66e66df23f8edb88112d3e152c779e0c38)
+++ src/main/java/ru/javawebinar/topjava/model/Meal.java	(date 1530137142426)
@@ -2,14 +2,14 @@
 
 import java.time.LocalDateTime;
 
-public class UserMeal {
+public class Meal {
     private final LocalDateTime dateTime;
 
     private final String description;
 
     private final int calories;
 
-    public UserMeal(LocalDateTime dateTime, String description, int calories) {
+    public Meal(LocalDateTime dateTime, String description, int calories) {
         this.dateTime = dateTime;
         this.description = description;
         this.calories = calories;
